src/pynxtools/dataconverter/helpers.py

```diff
@@ -685,28 +707,31 @@ def convert_str_to_bool_safe(value):
 def is_valid_data_field(value, nxdl_type, path):
     # todo: Check this funciton and wtire test for it. It seems the funciton is not
     # working as expected.
-    """Checks whether a given value is valid according to what is defined in the NXDL.
+    """Checks whether a given value is valid according to the type defined in the NXDL.
 
-    This function will also try to convert typical types, for example int to float,
-    and return the successful conversion.
+    This function also converts bool value comes in str format. In case, it fails to
+    convert, it raises an Exception.
 
-    If it fails to convert, it raises an Exception.
-
-    Returns two values: first, boolean (True if the the value corresponds to nxdl_type,
-    False otherwise) and second, result of attempted conversion or the original value
-    (if conversion is not needed or impossible)
+    Returns two values:
+        boolean (True if the the value corresponds to nxdl_type, False otherwise)
+        converted_value bool value.
     """
-    accepted_types = NEXUS_TO_PYTHON_DATA_TYPES[nxdl_type]
-    output_value = value
 
+    accepted_types = NEXUS_TO_PYTHON_DATA_TYPES[nxdl_type]
+    # Do not count the dict as it represents a link value
     if not isinstance(value, dict) and not is_valid_data_type(value, accepted_types):
         try:
             if accepted_types[0] is bool and isinstance(value, str):
                 value = convert_str_to_bool_safe(value)
                 if value is None:
                     raise ValueError
-            output_value = accepted_types[0](value)
-        except ValueError:
+                return True, value
+
+            collector.collect_and_log(
+                path, ValidationProblem.InvalidType, accepted_types, nxdl_type
+            )
+            return False, value
```

Comment:
This return value False here is interpreted as "undocumented field" in the calling function. So we get an additional wrong warning if the dtype does not match.

Reply:
I see that it is just for internal logic (please see line 750-766 in validation.py`. In `is_docmented` it checks the datatype, node type (is field or group). But the real type of the errors are collect and print according to the validation error type in and from `collect` function.

Reply:
I don't understand what you are trying to say here. My point is that is_documented returns False if the datatype does not match, which I would say is wrong, because it _is_ documented, but just has the wrong data type.

Reply:
This behavior was btw. introduced in https://github.com/FAIRmat-NFDI/pynxtools/pull/522, I believe. Maybe @GinzburgLev or @lukaspie can comment?

Reply:
I wanted to explain the code a bit, to clarify that in that code, no different type validation errors are collected, as you are assuming `undocumented field` warning will pop up.
By the way, I removed the second value from return statement as it is not being used anywhere.

Reply:
See #565